client_src/js/services/octoprint-client.service.js

```diff
@@ -0,0 +1,927 @@
+          UI.createAlert(
+            "error",
+            `There was an issue setting bed target attempt to ${printer.printerName} are you sure it's online?`,
+            3000,
+            "Clicked"
+          );
+        }
+      } else {
+        UI.createAlert(
+          "error",
+          `No response from ${printer.printerName}, is it online???`,
+          3000,
+          "Clicked"
+        );
+      }
+    }
+  }
+
+  static printerPreHeatTool(printer, toolTemp, toolNumber) {
+    let toolData = {
+      command: "target",
+      targets: {}
+    };
+    if (toolTemp.value !== "" && !isNaN(toolTemp.value)) {
+      toolData.targets["tool" + toolNumber.value] = parseInt(toolTemp.value);
+    }
+    if (toolTemp.value !== "" && !isNaN(toolTemp.value)) {
+      return OctoPrintClient.post(printer, "printer/tool", toolData);
+      if (typeof post !== "undefined") {
+        if (post.status === 204) {
+          UI.createAlert(
+            "success",
+            `Successfully set tool${toolNumber.value} target attempt to ${printer.printerName}...`,
+            3000,
+            "Clicked"
+          );
+        } else {
+          UI.createAlert(
+            "error",
+            `There was an issue setting tool${toolNumber.value} target attempt to ${printer.printerName} are you sure it's online?`,
+            3000,
+            "Clicked"
+          );
+        }
+      } else {
+        UI.createAlert(
+          "error",
+          `No response from ${printer.printerName}, is it online???`,
+          3000,
+          "Clicked"
+        );
+      }
+    }
+  }
+
+  static printerStartPrint(printer, e) {
+    e.target.disabled = true;
+    const opts = {
+      command: "start"
+    };
+    return OctoPrintClient.jobAction(printer, opts, e);
+  }
+
+  static printerPausePrint(printer, e) {
+    e.target.disabled = true;
+    const opts = {
+      command: "pause",
+      action: "pause"
+    };
+    return OctoPrintClient.jobAction(printer, opts, e);
+  }
+
+  static printerRestartPrint(printer, e) {
+    e.target.disabled = true;
+    const opts = {
+      command: "restart"
+    };
+    return OctoPrintClient.jobAction(printer, opts, e);
+  }
+
+  static printerResumePrint(printer, e) {
+    e.target.disabled = true;
+    const opts = {
+      command: "pause",
+      action: "resume"
+    };
+    return OctoPrintClient.jobAction(printer, opts, e);
+  }
+
+  static printerStopPrint(printer, e) {
+    e.target.disabled = true;
+    const opts = {
+      command: "cancel"
+    };
+    return OctoPrintClient.jobAction(printer, opts, e);
+  }
+
+  static printerMoveAxis(e, printer, axis, dir) {
+    return OctoPrintClient.move(e, printer, "jog", axis, dir);
+  }
+
+  static printerHomeAxis(e, printer, axis) {
+    return OctoPrintClient.move(e, printer, "home", axis);
+  }
+
+  static async octoPrintPluginInstallAction(printer, pluginList, action) {
+    let cleanAction = action.charAt(0).toUpperCase() + action.slice(1);
```

Comment:
WebStorm is indicating that this function (60 lines) is duplicate as its defined in printer-data as well